rispy/parser.py

```diff
@@ -197,27 +183,45 @@ def load(file, mapping=None, wok=False):
     of strings.
 
     Args:
-        file (object): File handle to read ris formatted data.
-        mapping (dict): Custom RIS tags mapping.
-        wok (bool): Use WOK format. Default False.
+        file (TextIO): File handle to read ris formatted data.
+        mapping (Dict, optional): a tag mapping dictionary.
+        is_wok (bool): Use WOK format. Default False.
 
     Returns:
         list: Returns list of RIS entries.
     """
     c = file.read()
+
     # Corrects for BOM in utf-8 encodings while keeping an 8-bit
     # string representation
-    if (c[0], c[1], c[2]) == ('\xef', '\xbb', '\xbf'):
+    if len(c) > 3 and (c[0], c[1], c[2]) == ("\xef", "\xbb", "\xbf"):
         c = c[3:]
 
-    return loads(c, mapping, wok)
+    return list(loads(c, mapping, is_wok))
+
+
+def loads(obj: str, mapping: Optional[Dict] = None, is_wok: bool = False) -> List[Dict]:
```

Comment:
There are plenty of RIS implementations. To make the API extensible, it might be a good idea to replace the boolean `is_wok` by something like `implementation="wok"` with `implementation=None` as the default (the default implementation).

Reply:
good suggestion; I'll make an enum w/ the default being base